app/assets/v2/js/grants/ingest-missing-contributions.js

```diff
@@ -26,30 +27,37 @@ Vue.component('grants-ingest-contributions', {
       let isValidTxHash;
       let isValidAddress;
 
-      // Validate that at least one of txHash and userAddress is provided
+      // Validate that only one of txHash and userAddress is provided. We only allow one because (1) that will be the
+      // most common use case, and (2) later when we update cart.js to fallback to manual ingestion if regular POSTing
+      // fails, that will make DRYing the manual ingestion code simpler. Reference implementation of DRY code for
+      // fallback to manual ingestion here: https://github.com/gitcoinco/web/pull/8563
       const { txHash, userAddress } = this.form;
-      const isFormComplete = txHash || userAddress;
+      const isFormComplete = Boolean(txHash) != Boolean(userAddress);
 
       if (!isFormComplete) {
         // Form was not filled out
-        this.$set(this.errors, 'invalidForm', 'Please enter a valid transaction hash or a valid wallet address');
+        this.$set(this.errors, 'invalidForm', 'Please enter a valid transaction hash OR a valid wallet address, but not both');
       } else {
         // Form was filled out, so validate the inputs
-        isValidTxHash = txHash && txHash.length === 66 && txHash.startsWith('0x');
+        isValidTxHash = txHash && txHash.length === 66 && ethers.utils.isHexString('0x');
```

Comment:
This will be true for any 66 char string - `... && ethers.utils.isHexString(txHash)`?

Reply:
Oops this is a typo, good catch. It should be `ethers.utils.isHexString(txHash)`

I'll fix this + @octavioamu's comment in the next commit, coming shortly...